docs/advanced/porting_guide.md

```diff
@@ -36,6 +36,52 @@ Add the target description to ```mbed-os\targets\targets.json```:
 
 See the [mbed Target Documentation](mbed_targets.md) for more details on what this definition means.
 
+### Coding requirements
+
+All code submitted as part of a port must compile with GCC ARM Embedded, ARM Compiler 5 and IAR EWARM. Flags that port code must compile with for each compiler are listed below. The profiles used to build mbed OS are subsets of each of these flags with exactly one optimization setting.
```

Comment:
For clarity, please change the second sentence to "Porting code must compile with specific flags for each compiler. See the list of these flags below." or something similar.

For clarity, please change the third sentence to "The profiles that build mbed OS are subsets of each of these flags. Each subset has one optimization setting." or something similar.